corehq/apps/groups/views.py

```diff
@@ -168,14 +167,21 @@ def _update_group_membership(request, domain, group_id):
     selected_users = request.POST.getlist('selected_ids')
 
     # check to make sure no users were deleted at time of making group
-    all_users = CommCareUser.ids_by_domain(domain)
-    safe_ids = [u for u in selected_users if u in all_users]
+    users = iter_docs(CouchUser.get_db(), selected_users)
+    safe_users = [
+        CouchUser.wrap_correctly(user) for user in users
+        if user['doc_type'] == 'CommCareUser' and user.get('domain') == domain
+    ]
+    safe_ids = [user.user_id for user in safe_users]
 
     group.users = safe_ids
 
     _ensure_case_sharing_privilege(request, group)
 
     group.save()
+
+    for user in safe_users:
+        update_user_in_es(None, user)
```

Comment:
We must have been doing this elsewhere as well, right? Is it clear what's broken with the current way of getting group changes into users' es docs? If not, I'm worried that that system may overwrite these changes if it's asynchronous and wrong, or generally that we'll have two systems to debug instead of one.

Reply:
The other place this should be happening is in the `group-pillow`: https://github.com/dimagi/commcare-hq/blob/f73f823081e15bb755f85321ffa104f2530e4243/corehq/pillows/groups_to_user.py#L100

I have not investigated to see if that's not working or perhaps just behind.

Reply:
@snopoke I think I figured it out in the commit I added to this PR. The group pillow make use of `group.removed_users` to figure out which users need to be updated, but this view was circumventing the code that adds to or removes from `removed_users`.

In the same PR I removed the line that eagerly puts them into elasticsearch. I wasn't totally sure about this, but I decided to do that so there's one clear way the information is supposed to make it to the User docs in ES.